xml-conduit/Text/XML/Stream/Render.hs

```diff
@@ -115,32 +117,53 @@ orderAttrs orderSpec = order
 -- the blaze-builder package, and allow the create of optimally sized
 -- 'ByteString's with minimal buffer copying.
 renderBuilder :: Monad m => RenderSettings -> Conduit Event m Builder
-renderBuilder RenderSettings { rsPretty = True, rsNamespaces = n, rsUseCDATA = useCDATA } = prettify =$= renderBuilder' n True useCDATA
-renderBuilder RenderSettings { rsPretty = False, rsNamespaces = n, rsUseCDATA = useCDATA } = renderBuilder' n False useCDATA
+renderBuilder settings@(RenderSettings { rsPretty = True }) = prettify =$= renderBuilder' settings
+renderBuilder settings@(RenderSettings { rsPretty = False }) = renderBuilder' settings
 
-renderBuilder' :: Monad m => [(Text, Text)] -> Bool -> (Content -> Bool) -> Conduit Event m Builder
-renderBuilder' namespaces0 isPretty useCDATA = do
+-- | Same as 'renderBuilder' but allows for flushing output when needed.
+--   Because it's guaranteed that 'Flush' would emit all the previously received data,
+--   this function doesn't support prettifying. Also it won't do certain other
+--   optimizations, like using self-closed tags.
+renderBuilderFlush :: Monad m => RenderSettings -> Conduit (Flush Event) m (Flush Builder)
+renderBuilderFlush settings@(RenderSettings { rsPretty = True }) = error "renderBuilderFlush: prettifying is not supported"
```

Comment:
Never mind, I read the comment. I see that there's a reason, I guess I'm just not following it. In any event, I think adding a comment explaining that `rsPretty` is ignored in this function would be preferable to errorring out.